### PR TITLE
Sanketika-Obsrv/issue-tracker#142: feat: scraping metrics with additional labels to prometheus

### PR DESCRIPTION
**Pull request by JeraldJF**

### CODE REVIEW

File: api-service/src/types/MetricModel.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export interface Labels {
+    entity: string,
+    id: string,
+    dataset_id: string | null,
+    endpoint: string,
+    status: number,
+    request_size: number,
+    response_size: string | number | string[] | undefined
```
Comment (HarishGangula) [inline api-service/src/types/MetricModel.ts:8]: why response_size is the not number ?

File: api-service/src/services/DatasetService.ts [hunk 2]
```diff
@@ -26,10 +27,12 @@ export const getDraftDataset = async (dataset_id: string) => {
...
-    return _.set(req, "id", apiId)
-}
-
 export const getDraftTransformations = async (dataset_id: string) => {
     return DatasetTransformationsDraft.findAll({ where: { dataset_id }, raw: true });
 }
+
+export const setReqDatasetId = (req: Request, dataset_id: string) => {
+    if (dataset_id) {
+        return _.set(req, "dataset_id", dataset_id)
```
Comment (HarishGangula) [inline api-service/src/services/DatasetService.ts:36]: if present set otherwise set to null

File: api-service/src/routes/Router.ts [hunk 3]
```diff
@@ -7,13 +7,16 @@ import DatasetRead from "../controllers/DatasetRead/DatasetRead";
...
 export const router = express.Router();
 
-router.post(`/v1/data/in/:datasetId`, setApiId("api.data.in"), dataIn);
-router.post('/v1/data/query', setApiId("api.data.out"), dataOut);
-router.post("/v1/datasets/create", setApiId("api.datasets.create"), DatasetCreate)
-router.patch("/v1/datasets/update", setApiId("api.datasets.update"), DatasetUpdate)
-router.get("/v1/datasets/read/:dataset_id", setApiId("api.datasets.read"), DatasetRead)
-router.post("/v1/datasets/list", setApiId("api.datasets.list"), DatasetList)
-router.get('/v1/data/exhaust/:datasetId', setApiId("api.data.exhaust"), dataExhaust);
+router.post(`/v1/data/in/:datasetId`, setApiId("api.data.in"),onRequest({ entity: "data-in" }), dataIn);
```
Comment (HarishGangula) [inline api-service/src/routes/Router.ts:15]: why not the enum defined entity types ?

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: api-service/src/services/DatasetService.ts (hunk 2 above)
Comment (JeraldJF) [reply api-service/src/services/DatasetService.ts:36]: Yes am setting it while getting the metric labels

const getMetricLabels = (req: any, res: Response) => {
     const { id, entity, url, startTime } = req;
     const { statusCode = 200 } = res
     const request_size = req.socket.bytesRead
     const response_size = res.getHeader('content-length');
     **const dataset_id = _.get(req, "dataset_id") || null**
     const duration = getDuration(startTime);
     const metricLabels = { entity, id, endpoint: url, dataset_id, status: statusCode, request_size, response_size }
     return { duration, metricLabels }
 }

File: api-service/src/types/MetricModel.ts (hunk 1 above)
Comment (JeraldJF) [reply api-service/src/types/MetricModel.ts:8]: There is a possibility that **res.getHeader('content-length')** method may return above types, but it's rare. Yesterday while testing all methods and cases, the number was returned.

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**